### ISSUE TITLE
Feature: default backup or not behaviour per namespace

### ISSUE THREAD

**Issue by dns2utf8** [vshn/k8up#73]
Dear vshn team

It would be very useful to have a control per namespace to set the default behaviour for the PVCs,
I am guessing this should be possible by modifying this section. 
https://github.com/vshn/k8up/blob/bf8c6386bb50d31a14aa73addffb6a30a5dd1c4e/service/backup/backupRunner.go#L116-L123

Cheers

**Comment by srueg** [comment]
What do you mean exactly with "default behaviour for the PVCs"?
Like defining to not back up PVCs by default (if they don't have the `ignore` annotation set)?

**Comment by dns2utf8** [comment]
Yes. For example, in the namespace `mongo-db-dev` people run the main pod `mongo-master` containing all the important data as well as a dynamic amount of `mongo-slave-XXX` to test the scaling.

In my opinion it were really nice to have one annotation in the namespace or it's schedule (eg. `appuio.ch/backup-pvc-default = false`) and one annotation on `mongo-master` with `appuio.ch/backup=true`.

**Comment by srueg** [comment]
Ok, cool. Yeah I also think that makes sense. Feel free to come up with a pull request, we're happy to support you.
I'll see if we can get around to implement such a feature :ok_hand:

**Comment by Kidswiss** [comment]
We have an internal ticket for implementing label selectors for the job CRDs.

The idea is that you could define a label selector for a job object which will only target the PVCs or pods with the label.

That should also cover your use-case, right? @dns2utf8 That should give even more control over what should get backed up per schedule object.

**Comment by dns2utf8** [comment]
Yes such a selector in the schedule object would be very useful.
It would also solve the problem with temporary Pods/PVCs created by other systems.

**Comment by Kidswiss** [comment]
dang I thought I linked to the docs :) That selector already exists: `appuio.ch/backup=false`

See the getting started tutorial: https://k8up.io/docs/0.1.5/getting-started.html

**Comment by dns2utf8** [comment]
I am using `appuio.ch/backup=false` for more my deployments already.
However, some applications create volumes on the go for example to store intermediate results during processing. It is very hard/impossible to add the annotation to theses PVCs when I don't build that container myself.
The same is true for helm charts that recreate PVCs for eg. redis Pods.

**Comment by Kidswiss** [comment]
Is it possible to move these things into separate namespaces then? That don't contain a backup schedule at all? E.g. prometheus to a monitoring namespace, appX in the appXnamespace. etc.

We're internally discussing how to get better selection options into k8up. Something in the likes of a `backupTemplate` where you can define an array of globs that should be excluded from the backups instead of the annotations.
